build/version.go

```diff
@@ -0,0 +1,59 @@
+package build
+
+import (
+	"strconv"
+	"strings"
+)
+
+// Version is the current version of siad.
+const Version = "0.3.3.1"
+
+// IsVersion returns whether str is a valid version number.
+func IsVersion(str string) bool {
+	for _, n := range strings.Split(str, ".") {
+		if _, err := strconv.Atoi(n); err != nil {
+			return false
+		}
+	}
+	return true
+}
+
+// min returns the smaller of two integers.
+func min(a, b int) int {
+	if a < b {
+		return a
+	}
+	return b
+}
+
+// VersionCmp returns an int indicating the difference between a and b. It
+// follows the convention of bytes.Compare and big.Cmp:
+//
+//   -1 if a <  b
+//    0 if a == b
+//   +1 if a >  b
+//
+// One important quirk is that "1.1.0" is considered newer than "1.1", despite
+// being numerically equal.
```

Comment:
it's good that the quirk is documented, but we should never have a release that goes 1.1 -> 1.1.0 nor 1.1.0 -> 1.1